code/modules/antagonists/disease/disease_mob.dm

```diff
@@ -0,0 +1,318 @@
+/*
+A mob of type /mob/camera/disease is an overmind coordinating at least one instance of /datum/disease/advance/sentient_disease
+that has infected a host. All instances in a host will be synchronized with the stats of the overmind's disease_template. Any
+samples outside of a host will retain the stats they had when they left the host, but infecting a new host will cause
+the new instance inside the host to be updated to the template's stats.
+*/
+
+/mob/camera/disease
+	name = ""
+	real_name = ""
+	desc = ""
+	icon = 'icons/mob/blob.dmi'
+	icon_state = "marker"
+	mouse_opacity = MOUSE_OPACITY_ICON
+	move_on_shuttle = 1
+	see_in_dark = 8
+	invisibility = INVISIBILITY_OBSERVER
+	layer = BELOW_MOB_LAYER
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	sight = SEE_SELF
+	initial_language_holder = /datum/language_holder/empty
+
+	var/datum/action/innate/disease_adapt/adaptation_menu_action
+	var/datum/disease_ability/examining_ability
+	var/datum/browser/browser
+	var/browser_open = FALSE
+
+	var/mob/living/following_host
+	var/datum/component/redirect/move_listener
+	var/list/disease_instances
+	var/list/hosts //this list is associative, affected_mob -> disease_instance
+	var/datum/disease/advance/sentient_disease/disease_template
+
+	var/total_points = 0
+	var/points = 0
+
+	var/last_move_tick = 0
+	var/move_delay = 1
+
+	var/next_adaptation_time = 0
+	var/adaptation_cooldown = 1200
+
+	var/list/purchased_abilities
+	var/list/unpurchased_abilities
+
+/mob/camera/disease/Initialize(mapload)
+	.= ..()
+	adaptation_menu_action = new /datum/action/innate/disease_adapt()
+	adaptation_menu_action.Grant(src)
+
+	disease_instances = list()
+	hosts = list()
+
+	purchased_abilities = list()
+	unpurchased_abilities = list()
+	for(var/V in GLOB.disease_ability_singletons)
+		unpurchased_abilities[V] = TRUE
+		var/datum/disease_ability/A = V
+		if(A.start_with && A.CanBuy(src))
+			A.Buy(src, TRUE, FALSE)
+
+	disease_template = new /datum/disease/advance/sentient_disease()
+	disease_template.overmind = src
+	qdel(SSdisease.archive_diseases[disease_template.GetDiseaseID()])
+	SSdisease.archive_diseases[disease_template.GetDiseaseID()] = disease_template //important for stuff that uses disease IDs
+
+	var/datum/atom_hud/my_hud = GLOB.huds[DATA_HUD_SENTIENT_DISEASE]
+	my_hud.add_hud_to(src)
+
+	browser = new /datum/browser(src, "disease_menu", "Adaptation Menu", 1000, 770, src)
+
+/mob/camera/disease/Destroy()
+	. = ..()
+	for(var/V in GLOB.sentient_disease_instances)
+		var/datum/disease/advance/sentient_disease/S = V
+		if(S.overmind == src)
+			S.overmind = null
+
+/mob/camera/disease/Stat()
+	..()
+	if(statpanel("Status"))
+		stat("Adaptation Points: [points]/[total_points]")
+		stat("Hosts: [disease_instances.len]")
+		var/adapt_ready = next_adaptation_time - world.time
+		if(adapt_ready > 0)
+			stat("Adaptation Ready: [round(adapt_ready/10, 0.1)]s")
+
+/mob/camera/disease/say(message)
+	return
+
+/mob/camera/disease/Move(NewLoc, Dir = 0)
```

Comment:
I feel this would make more sense on key focus level but that can be changed later.